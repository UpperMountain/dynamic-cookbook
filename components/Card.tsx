import React from "react";
import { View, ViewProps } from "react-native";
import Padded from "./Padded";
import shadow from "../lib/shadow";

export interface Props {
  children: React.ReactNode;

  // passed to shadow()
  shadowAmt?: number;

  // Container View style overrides
  style?: ViewProps["style"];

  // Inner view style overrides
  innerStyle?: ViewProps["style"];
}

export default function Card(props: Props) {
  const { children, shadowAmt = 0.1, style, innerStyle } = props;
  return (
    <Padded bottom style={style}>
      <View
        style={[
          {
            backgroundColor: "white",
            ...shadow(shadowAmt)
          },
          innerStyle
        ]}
      >
        {children}
      </View>
    </Padded>
  );
}
