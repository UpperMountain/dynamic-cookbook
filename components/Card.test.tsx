import React from "react";
import { Text } from "react-native";
import Card from "./Card";
import Renderer from "react-test-renderer";

it("should render without exploding", () => {
  Renderer.create(
    <Card>
      <Text>test</Text>
    </Card>
  ).unmount();
});

it("should match snapshot", () => {
  const cmp = Renderer.create(
    <Card
      style={{ backgroundColor: "red" }}
      innerStyle={{ backgroundColor: "blue" }}
      shadowAmt={0.5}
    >
      <Text>test</Text>
    </Card>
  );
  expect(cmp.toJSON()).toMatchSnapshot();
  cmp.unmount();
});
