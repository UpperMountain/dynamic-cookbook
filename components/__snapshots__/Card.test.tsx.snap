// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "paddingBottom": 20,
        "paddingLeft": 0,
        "paddingRight": 0,
        "paddingTop": 0,
      },
      Object {
        "backgroundColor": "red",
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "white",
          "elevation": 12,
          "shadowColor": "#000",
          "shadowOffset": Object {
            "height": 5.5,
            "width": 0,
          },
          "shadowOpacity": 0.375,
          "shadowRadius": 10,
        },
        Object {
          "backgroundColor": "blue",
        },
      ]
    }
  >
    <Text>
      test
    </Text>
  </View>
</View>
`;
